presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -952,6 +955,25 @@ The following procedures are available:
 
   Invalidate directory list cache for specified directory_path.
 
+* ``system.invalidate_metastore_cache()``
+
+  Invalidate all metastore caches.
+
+* ``system.invalidate_metastore_cache(schema_name)``
+
+  Invalidate all metastore cache entries linked to a specific schema.
+
+* ``system.invalidate_metastore_cache(schema_name, table_name)``
+
+  Invalidate all metastore cache entries linked to a specific table.
+
+* ``system.invalidate_metastore_cache(schema_name, table_name, partition_columns, partition_values)``
```

Comment:
For partitioned table, does it make sense to invalidate the entire metastore cache when adding/deleting partitions or use of sync partition procedure make more sense? if latter, better off to put a note to link to ``system.sync_partition_metadata``

Reply:
The approach will depend on the specific use case. If the partitions have already been added to the metastore, we will need to use cache invalidation. However, if the partitions have not yet been added to the metastore, we should use the `sync_partition_metadata` procedure.

Therefore, I would say that the two methods are not related, since cache invalidation is only useful if the partitions are already synced to the metastore but are not yet visible in Presto.